pkg/endpoint/bpf.go

```diff
@@ -1072,58 +1064,66 @@ func (e *Endpoint) applyPolicyMapChanges(regenContext *regenerationContext, hasN
 	return nil
 }
 
-// syncPolicyMapWith updates the bpf policy map state based on the
-// difference between the given 'realized' and desired policy state without
+// syncRealizedPolicyMap updates the bpf policy map state based on the
+// difference between the realized and desired policy state without
 // dumping the bpf policy map.
-// Changes are synced to endpoint's realized policy mapstate, 'realized' is
-// not modified.
-func (e *Endpoint) syncPolicyMapWith(realized policy.MapState, withDiffs bool) (diffCount int, diffs []policy.MapChange, err error) {
+// Only called when desired and realized policies are not the same.
+func (e *Endpoint) syncRealizedPolicyMap() error {
+	errors := 0
+
+	// Add policy map entries before deleting to avoid transient drops
```

Comment:
@nathanjsweet realized we need to handle the case where add-then-delete pushes us over the mapstate limit, even if the end-result will fit.

So, you can either compute the sizes beforehand, or detect an oversize error, and re-try adds after deletes in that case.

Reply:
Added a commit on the latter strategy.